src/runtime_src/core/common/device.h

```diff
@@ -259,6 +259,28 @@ class device : public ishim
     return reinterpret_cast<SectionType>(get_axlf_section_or_error(section, xclbin_id).first);
   }
 
+  /**
+   * get_axlf_sections() - Get sections from currently loaded axlf
+   *
+   * xclbin_id:  Check that xclbin_id matches currently cached
+   * Return:     Vectors of Pair of section data and size in bytes
+   *
+   * This function provides access to meta data sections that are
+   * from currently loaded xclbin.  The returned sections are from when the
+   * xclbin was loaded by this process.  The function cannot be used
+   * unless this process loaded the xclbin.
+   *
+   * The function returns {nullptr, 0} if section is not cached.
+   *
+   * Same behavior as other get_axlf_sections()
+   */
+  XRT_CORE_COMMON_EXPORT
+  std::vector<std::pair<const char*, size_t> >
+  get_axlf_sections(axlf_section_kind section, const uuid& xclbin_id = uuid()) const;
```

Comment:
The API specifies a single section, why is it returning a vector of sections?
For all usage of templated types, please remove the extra space between '> >',  this is very old C++ synatx.